### PR TITLE
Small docs fixes

### PR DESCRIPTION
**Pull request by nvaytet**
- remove deprecation warnings in plotting notebooks
- attempt to fix raw matplotlib figures in docs (see double figures in https://scipp.github.io/user-guide/binned-data/binned-data.html)

### CODE REVIEW

File: docs/visualization/plotting-1d-data.ipynb [hunk 1]
```diff
@@ -39,7 +39,7 @@
    "source": [
     "d = sc.Dataset()\n",
     "N = 50\n",
-    "d.coords['tof'] = sc.Variable(['tof'], values=np.arange(N).astype(np.float64),\n",
+    "d.coords['tof'] = sc.Variable(['tof'], values=np.arange(N, dtype=np.float64),\n",
```
Comment (OwenArnold) [inline docs/visualization/plotting-1d-data.ipynb:42]: Very minor point, we could also move the dtype specification to the `Variable` itself. I'm not sure if that has any major benefit other than showing that we have dtype control at the point of variable creation.

File: docs/visualization/plotting-1d-data.ipynb [hunk 2]
```diff
@@ -413,7 +413,7 @@
     "d4['Sample'].masks['mask1'] = sc.Variable(dims=['tof'],\n",
     "                                          values=np.where(np.abs(x-40) < 10, True, False))\n",
     "d4['Background'].masks['mask1'] = ~d4['Sample'].masks['mask1']\n",
-    "d4['Background'].masks['mask1']['tof', 0:20].values = np.zeros(20, dtype=np.bool)\n",
+    "d4['Background'].masks['mask1']['tof', 0:20].values = np.zeros(20, dtype=bool)\n",
```
Comment (OwenArnold) [inline docs/visualization/plotting-1d-data.ipynb:416]: Also, numpy is now deprecating the use of it's internal types `np.bool` -> `bool`.  FYI https://numpy.org/devdocs/release/1.20.0-notes.html

Comment (nvaytet) [reply docs/visualization/plotting-1d-data.ipynb:416]: yes, that is the reason for this change

Comment (OwenArnold) [reply docs/visualization/plotting-1d-data.ipynb:416]: 😆 I need to read more carefully

**Comment by nvaytet** [comment]
Closing in favour of #1882

**Pull request closed without merging**